src/service/extensions-impl.js

```diff
@@ -203,17 +196,16 @@ export class Extensions {
    * loading/registration.
    * @param {string} extensionId
    * @param {string} version
-   * @param {number=} opt_timeout
    * @return {!Promise<?ExtensionDef>}
    */
-  waitForExtension(extensionId, version, opt_timeout) {
-    return /** @type {!Promise<?ExtensionDef>} */ (Services.timerFor(
-      this.win
-    ).timeoutPromise(
-      opt_timeout || LOAD_TIMEOUT,
-      this.waitFor_(this.getExtensionHolder_(extensionId, version)),
-      `Render timeout waiting for extension ${extensionId} to be load.`
-    ));
+  waitForExtension(extensionId, version) {
+    const wait = this.waitFor_(this.getExtensionHolder_(extensionId, version));
+    const timeout = new Promise((r) => this.win.setTimeout(r, 16000)).then(
+      () => {
+        user().error(TAG, `Waited over 16s to load extension ${extensionId}.`);
+      }
+    );
```

Comment:
This will _always_ fire the error after 16s. You need:

```js
const wait = this.waitFor_();
return timeoutPromise(16000, wait).catch(() => wait)
```

Reply:
Great catch!